streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -5392,6 +5392,31 @@ object ZStreamSpec extends ZIOBaseSpec {
             }
           }
         ),
+        suite("fromInputStreamInterruptible")(
+          test("should read all data from InputStream correctly") {
+            val chunkSize = ZStream.DefaultChunkSize
+            val data      = Array.tabulate[Byte](chunkSize * 5 / 2)(_.toByte)
+            def is        = new ByteArrayInputStream(data)
+            ZStream.fromInputStreamInterruptible(is, chunkSize).runCollect map { bytes =>
+              assert(bytes.toArray)(equalTo(data))
+            }
+          },
+          test("ZStream.fromInputStreamInterruptible should handle interruption") {
+            for {
+              latch      <- Promise.make[Nothing, Unit]
+              data       <- ZIO.succeed("Interruptible Stream!".getBytes("UTF-8"))
+              inputStream = new ByteArrayInputStream(data)
+              fiber <- ZStream
+                         .fromInputStreamInterruptible(inputStream)
+                         .tap(_ => latch.succeed(()))
```

Comment:
This is going to be flaky due to a race condition. Use this instead 
```suggestion
                         .tap(_ => latch.succeed(()) *> ZIO.never)
```